app/controllers/free_item_requests_controller.rb

```diff
@@ -0,0 +1,24 @@
+class FreeItemRequestsController < ApplicationController
+  before_action :find_free_item
+
+  def new
+    @request = FreeItemRequest.create(item: @item, actual_owner: @item.user, potential_owner: current_user)
+    redirect_to root_path, notice: "You've sent request to this item's user."
+  end
+
+  def give_away
+    @request = FreeItemRequest.find_by_id(params[:id])
+    Item.create(title: @item.title, description: @item.description, user: @request.potential_owner, free: false)
+    @item.destroy
+    @request.destroy
+    redirect_to root_path, notice: "Congrats, you're a good person!"
+  end
+
+  def filter_notifications
+    @requests = current_user.pending_requests
+  end
+
+  def find_free_item
```

Comment:
So usually this method is called set_<bla-bla-bla>.
Also to avoid later problems with nil, I suggest to use find method.